src/GitHub.App/Controllers/NavigationController.cs

```diff
@@ -66,7 +66,15 @@ public void LoadView(IConnection connection, ViewWithData data, Action<IView> on
             switch (data.MainFlow)
             {
                 case UIControllerFlow.PullRequestCreation:
+                    //The selected flow will change after the view is created, so save it.
+                    var previousFlow = Current?.SelectedFlow;
+
                     CreateView(connection, data, onViewLoad);
+                    if (data.Data == null && previousFlow == UIControllerFlow.PullRequestCreation)
+                    {
+                        Pop(history[history.Count-2]);
```

Comment:
I agree! As it stands, there will always be at least two items in the history, since you can't get to the PR creation view without going to the PR list view first. That may change over time so it's best to add that assert in case we forget when there is more than one way to get to the creation view!